AdyenActions/Components/3DS2/Action handlers/3DS2+Delegated Authentication/ThreeDS2PlusDACoreActionHandler.swift

```diff
@@ -148,6 +148,108 @@
+                                                                  failureHandler: failureHandler)
+                },
+                notRegisteredHandler: failureHandler
+            )
+        }
+        
+        // MARK: Delegated Authentication Approval
+        
+        private func showApprovalScreenWhenDeviceIsRegistered(delegatedAuthenticationInput: String,
+                                                              completion: @escaping (Result<String, DelegateAuthenticationError>) -> Void,
+                                                              failureHandler: @escaping () -> Void) {
+            presenter.showApprovalScreen(component: self,
+                                         approveAuthenticationHandler: { [weak self] in
+                                             guard let self = self else { return }
+                                             self.executeDAAuthenticate(delegatedAuthenticationInput: delegatedAuthenticationInput,
+                                                                        authenticatedHandler: { completion(.success($0)) },
+                                                                        failedAuthenticationHandler: failureHandler)
+                                         },
+                                         fallbackHandler: {
+                                             failureHandler()
+                                         },
+                                         removeCredentialsHandler: { [weak delegatedAuthenticationService] in
+                                             try? delegatedAuthenticationService?.reset()
+                                             failureHandler()
+                                         })
+        }
+
+        private func executeDAAuthenticate(delegatedAuthenticationInput: String,
+                                           authenticatedHandler: @escaping (String) -> Void,
+                                           failedAuthenticationHandler: @escaping () -> Void) {
+            delegatedAuthenticationService.authenticate(withAuthenticationInput: delegatedAuthenticationInput) { result in
+                switch result {
+                case let .success(sdkOutput):
+                    authenticatedHandler(sdkOutput)
+                case .failure:
+                    failedAuthenticationHandler()
+                }
+            }
+        }
+        
+        private func isDeviceRegisteredForDelegatedAuthentication(delegatedAuthenticationInput: String,
+                                                                  registeredHandler: @escaping () -> Void,
+                                                                  notRegisteredHandler: @escaping () -> Void) {
+            delegatedAuthenticationService.isDeviceRegistered(withAuthenticationInput: delegatedAuthenticationInput) { result in
+                switch result {
+                case .failure:
+                    notRegisteredHandler()
```

Comment:
the underlying error here is ignored, it should be added as the cause error

Reply:
Done. Thanks ✅